---
title: "Results"
execute: 
  warning: false
  error: false
output:
  html_document:
    toc: true
    toc_depth: 4
    theme: bootstrap
    highlight: breezedark
    code-fold: true
---


## Libraries and Packages

```{r, warning=FALSE, message=FALSE}
library(tidyverse)
library(knitr)
```

```{r, eval=FALSE}
if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

BiocManager::install("ggtree")
```




```{r}
library(tidyverse)
library(ggtree)
# added treeio and ggimage library
library(treeio)
library(ggimage)
library(rphylopic)
```

```{r}
library(tidyverse)
library(knitr)
library(ggtree)
library(TDbook) #A Companion Package for the Book "Data Integration, Manipulation and Visualization of Phylogenetic Trees" by Guangchuang Yu (2022, ISBN:9781032233574).
library(ggimage)
library(rphylopic)
library(treeio)
library(tidytree)
library(ape)
library(TreeTools)
library(phytools)
library(ggnewscale)
library(ggstar)
```
```{r}
if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

BiocManager::install("Biostrings")
```
```{r, eval=FALSE}
install.packages("boot")
install.packages("codetools")
install.packages("lattice")
install.packages("survival")
```

## Loaded Data 

```{r}
NEON_MAGs <- read_csv("data/NEON/GOLD_Study_ID_Gs0161344_NEON_edArchaea.csv") %>% 
  # remove columns that are not needed for data analysis
  select(-c(`GOLD Study ID`, `Bin Methods`, `Created By`, `Date Added`)) %>% 
  # create a new column with the Assembly Type
  mutate("Assembly Type" = case_when(`Genome Name` == "NEON combined assembly" ~ `Genome Name`,
                            TRUE ~ "Individual")) %>% 
  mutate_at("Assembly Type", str_replace, "NEON combined assembly", "Combined") %>% 
  separate(`GTDB-Tk Taxonomy Lineage`, c("Domain", "Phylum", "Class", "Order", "Family", "Genus"), "; ", remove = FALSE) %>% 
  # Get rid of the the common string "Soil microbial communities from "
  mutate_at("Genome Name", str_replace, "Terrestrial soil microbial communities from ", "") %>% 
  # Use the first `-` to split the column in two
  separate(`Genome Name`, c("Site","Sample Name"), " - ") %>% 
  # Get rid of the the common string "S-comp-1"
  mutate_at("Sample Name", str_replace, "-comp-1", "") %>%
  # separate the Sample Name into Site ID and plot info
  separate(`Sample Name`, c("Site ID","subplot.layer.date"), "_", remove = FALSE,) %>% 
  # separate the plot info into 3 columns
  separate(`subplot.layer.date`, c("Subplot", "Layer", "Date"), "-") 
```

```{r}
NEON_metagenomes <- read_tsv("data/NEON/exported_img_data_Gs0161344_NEON.tsv") %>% 
  rename(`Genome Name` = `Genome Name / Sample Name`) %>% 
  filter(str_detect(`Genome Name`, 're-annotation', negate = T)) %>% 
  filter(str_detect(`Genome Name`, 'WREF plot', negate = T)) 
```

```{r}
NEON_metagenomes <- NEON_metagenomes %>% 
  # Get rid of the the common string "Soil microbial communities from "
  mutate_at("Genome Name", str_replace, "Terrestrial soil microbial communities from ", "") %>% 
  # Use the first `-` to split the column in two
  separate(`Genome Name`, c("Site","Sample Name"), " - ") %>% 
  # Get rid of the the common string "-comp-1"
  mutate_at("Sample Name", str_replace, "-comp-1", "") %>%
  # separate the Sample Name into Site ID and plot info
  separate(`Sample Name`, c("Site ID","subplot.layer.date"), "_", remove = FALSE,) %>% 
  # separate the plot info into 3 columns
  separate(`subplot.layer.date`, c("Subplot", "Layer", "Date"), "-") 
```
```{r}
NEON_chemistry <- read_tsv("data/NEON/neon_plot_soilChem1_metadata.tsv") %>% 
  # remove -COMP from genomicsSampleID
  mutate_at("genomicsSampleID", str_replace, "-COMP", "") 
```

```{r}
NEON_MAGs_metagenomes_chemistry <- NEON_MAGs %>% 
  left_join(NEON_metagenomes, by = "Sample Name") %>%
  left_join(NEON_chemistry, by = c("Sample Name" = "genomicsSampleID"))
head(NEON_MAGs_metagenomes_chemistry)
```
```{r}
NEON_MAGs_metagenomes_chemistry_alphaproteobacteria <- NEON_MAGs_metagenomes_chemistry %>% 
  left_join(NEON_metagenomes, by = "Sample Name") %>%
  left_join(NEON_chemistry, by = c("Sample Name" = "genomicsSampleID")) %>%
  filter(`Class`== "Alphaproteobacteria")
head(NEON_MAGs_metagenomes_chemistry_alphaproteobacteria)
```
```{r}
NEON_MAGs_metagenomes_chemistry_alphaproteobacteria %>%  
ggplot(aes(x = `Site.x`, y = soilTemp.x)) +
  geom_boxplot()
```

```{r}
NEON_chemistry_select <- NEON_chemistry %>%
  select(`genomicsSampleID`, `siteID`, `nlcdClass`)
head(NEON_chemistry_select)
```
```{r}
NEON_MAGs_select <- NEON_MAGs %>%
  select(`Sample Name`, `Site ID`, `GTDB-Tk Taxonomy Lineage`)
  head(NEON_MAGs)
```

```{r}
filtered_data <- NEON_MAGs_metagenomes_chemistry %>% 
  filter(`Site.x` == "Chase Lake Wetlands")
```





```{r}
NEON_MAGs_select_WOOD <- NEON_MAGs_select %>%
  filter(`Site ID`== "WOOD")
  head(NEON_MAGs_select_WOOD)
```
```{r}
NEON_metagenomes_select <- NEON_metagenomes %>%
  select(`Sample Name`, `Site ID`, `Ecosystem Subtype`)
head(NEON_metagenomes_select)
```

```{r}
NEON_metagenomes_select_WOOD <- NEON_MAGs_metagenomes_chemistry %>%
  filter(`Site ID.x`== "WOOD")
  head(NEON_metagenomes_select_WOOD)
```
```{r}
NEON_metagenomes_select_WOOD %>%  
ggplot(aes(x = `Site.x`, y = soilTemp)) +
  geom_boxplot()
```

```{r}
NEON_metagenomes_select_WOOD %>% 
  left_join(NEON_MAGs_select_WOOD, by = "Sample Name")
```
## Graphs

### Lab 12

```{r}
NEON_MAGs_metagenomes_chemistry_alphaproteobacteria %>%  
ggplot(aes(x = `soilTemp.x`, y = Site.x)) +
  geom_boxplot()
```



```{r}
ggplot(NEON_MAGs_metagenomes_chemistry_alphaproteobacteria, aes(x = `soilTemp.x`, y = `Ecosystem Subtype.x`, color = Class)) +
  geom_point() +
  labs(title = "Scatterplot of Ecosystem Subtype vs Soil Temperature", 
       x = "soilTemp", 
       y = "Ecosystem Subtype",
       color = "Order")
```

```{r}
NEON_MAGs_select_WOOD %>% 
  ggplot(aes(x = `Site ID`, y = `GTDB-Tk Taxonomy Lineage`)) +
  geom_point()
```

```{r}
ggplot(NEON_MAGs_metagenomes_chemistry_alphaproteobacteria, aes(x = `Site.x`, y = `Geographic Location.x`, fill = Family)) +
  geom_boxplot() +
  labs(title = "Violin Plot of Geographic Location by Site",
       x = "Site",
       y = "Geographic Location",
       fill = "Family") +
  scale_x_discrete(labels = abbreviate) +
  theme(axis.text.x = element_text(angle = 60, hjust = 1, vjust = 1, size = 8))
```
```{r}
NEON_MAGs_metagenomes_chemistry_alphaproteobacteria %>%  
ggplot(aes(x = `Site.x`, y = `Geographic Location.x`)) +
  geom_boxplot()
```
```{r}
NEON_MAGs_metagenomes_chemistry_alphaproteobacteria %>%  
  ggplot(aes(x = `Site.x`, y = `Geographic Location.x`)) +
  geom_boxplot() +
  labs(
    title = "Boxplot of Geographic Location by Site",
    x = "Site",
    y = "Geographic Location"
  ) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
```



